deployments/quick-start/kubernetes-vault-rbac.yaml

```diff
@@ -0,0 +1,110 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: kube-vault
+  namespace: default
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: kube-vault
+rules:
+- apiGroups: [""]
+  resources:
+  - pods
+  verbs: ["get", "list"]
```

Comment:
The `kube.go` file is correct. Sorry, I forgot that it does list pods, so in this case, we do need to have the `list` permission for pods. 

For endpoints, I think the verb we need is `read`. My apologies for getting them confused!

In general cases, I think most people will want to give their `kubernetes-vault` permission to watch all namespaces and I the changes you've made in this PR should be sufficient. I'd say people would need to roll their own specific `kubenetes-vault.yaml` for prod anyway.

However with RBAC, I am wondering if kubernetes-vault would still work correctly (unrelated to this PR) if kubernetes-vault is restricted to a certain namespace using rbac.

One possible scenario might be the following:
- I have a small kube deployment with the `staging` and `prod` namespaces.
- I want to have 2 instances of `kubernetes-vault`, 1 in `staging` and 1 in `prod`.
- If the `prod` `kubernetes-vault` instance is restricted to the `prod` namespace, will the call to `/api/v1/watch/pods` work correctly?